ert3/storage/_storage.py

```diff
@@ -133,9 +180,81 @@ def _get_data(workspace: Path, experiment_name: str, record_name: str) -> Any:
     if response.status_code == 404:
         raise KeyError(f"No {record_name} data for experiment: {experiment_name}")
 
+    if response.status_code != 200:
+        raise ert3.exceptions.StorageError(response.text)
+
     return response.content
 
 
+def _response2record(
+    response_content: bytes, record_type: ert3.data.RecordType, realization_id: int
+) -> ert3.data.Record:
+    dataframe = pd.read_csv(
+        io.BytesIO(response_content), index_col=0, float_precision="round_trip"
+    )
+
+    raw_index = tuple(dataframe.columns)
+    if record_type == ert3.data.RecordType.LIST_FLOAT:
+        array_data = tuple(
+            float(dataframe.loc[realization_id][raw_idx]) for raw_idx in raw_index
+        )
+        return ert3.data.Record(data=array_data)
+    elif record_type == ert3.data.RecordType.MAPPING_INT_FLOAT:
+        int_index = tuple(int(e) for e in dataframe.columns)
+        idata = {
+            idx: float(dataframe.loc[realization_id][raw_idx])
+            for raw_idx, idx in zip(raw_index, int_index)
+        }
+        return ert3.data.Record(data=idata)
+    elif record_type == ert3.data.RecordType.MAPPING_STR_FLOAT:
+        str_index = tuple(str(e) for e in dataframe.columns)
+        sdata = {
+            idx: float(dataframe.loc[realization_id][raw_idx])
+            for raw_idx, idx in zip(raw_index, str_index)
+        }
+        return ert3.data.Record(data=sdata)
+    else:
+        raise ValueError(
+            f"Unexpected record type when loading numerical record: {record_type}"
+        )
+
+
+def _get_numerical_data(
+    workspace: Path, experiment_name: str, record_name: str
+) -> ert3.data.EnsembleRecord:
+    experiment = _get_experiment_by_name(experiment_name)
+    if experiment is None:
+        raise KeyError(
+            f"Cannot get {record_name} data, no experiment named: {experiment_name}"
+        )
+
+    ensemble_id = experiment["ensembles"][0]  # currently just one ens per exp
+
+    metadata = json.loads(
+        _get_file_data(workspace, experiment_name, record_name + _METADATA)
+    )
+    record_type = ert3.data.RecordType(metadata["record_type"])
+
+    records = []
+    for real_id in range(metadata["ensemble_size"]):
+        response = requests.get(
+            url=f"{_STORAGE_URL}/ensembles/{ensemble_id}/records/{record_name}",
+            params={"realization_index": real_id},
+            headers={"accept": "application/x-dataframe"},
```

Comment:
Should you pin ert storage note that you use this deprecated content type?

Reply:
`ert-storage` is already pinned to an exact version and in that version (which is the latest tagged version) this is not a deprecated content type nor is the content type currently encouraged on master available 🤷🏻 I can for sure write a comment somewhere, but I'm thinking that this is something that will bubble up when moving the `ert-storage` tag anyway...